pylint/lint/pylinter.py

```diff
@@ -602,32 +604,26 @@ def process_tokens(self, tokens):
 
     # code checking methods ###################################################
 
-    def get_checkers(self):
+    def get_checkers(self) -> list[BaseChecker]:
         """Return all available checkers as a list."""
-        return [self] + [
-            c
-            for _checkers in self._checkers.values()
-            for c in _checkers
-            if c is not self
-        ]
+        return sorted(c for _checkers in self._checkers.values() for c in _checkers)
 
-    def get_checker_names(self):
+    def get_checker_names(self) -> list[str]:
         """Get all the checker names that this linter knows about."""
-        current_checkers = self.get_checkers()
         return sorted(
             {
                 checker.name
-                for checker in current_checkers
+                for checker in self.get_checkers()
                 if checker.name != MAIN_CHECKER_NAME
             }
         )
```

Comment:
We do need to sort again as the normal sorting order takes into account whether something is an `extension`. For some reason `get_checkers` without my fix to `__gt__` actually broke a couple of tests so it seems to order of `get_checker_names` and `get_checkers` is important. I'd say let's keep sorting on `.name` (and thereby forgetting about `extensions`) here.

Reply:
It could be important, or it could be tests that all the element of the list exists and are the same in the expected data. Ordering permits to do ``my_list == expected`` which is simpler than ``all(a in expected for a in my_list) and all(b in my_list for b in expected)`` (or ``sorted(my_list) == sorted(expected)``) ?

Reply:
It were a couple (5/10) functional tests that broke. So I'm leaning towards internal behaviour that we don't know about 😅

Reply:
Ho well, if it's functional tests let's not bother.